iroh/src/p2p.rs

```diff
@@ -49,13 +49,23 @@ impl FromStr for PeerIdOrAddrArg {
 
 pub async fn run_command(p2p: &impl P2pApi, cmd: &P2p) -> Result<()> {
     match &cmd.command {
-        P2pCommands::Connect { .. } => {
-            todo!("`iroh p2p connect` is not yet implemented")
-        }
+        P2pCommands::Connect { addr } => match p2p.connect(&addr.0).await {
+            Ok(_) => {
+                println!("Connected!");
+            }
+            Err(e) => return Err(e),
+        },
         P2pCommands::Lookup { addr } => {
-            let lookup = p2p.lookup(&addr.0).await?;
-            println!("peer id: {}", lookup.peer_id);
+            let lookup = match addr {
+                Some(addr) => p2p.lookup(&addr.0).await?,
+                None => p2p.lookup_local().await?,
+            };
+            display_lookup(&lookup);
         }
     };
     Ok(())
 }
+
+fn display_lookup(l: &Lookup) {
+    println!("peer id: {}\nlistening addresses: {:?}\nprotocols: {:?}\nprotocol version: {}\nobserved addresses: {:?}", l.peer_id, l.listen_addrs, l.protocols, l.protocol_version, l.observed_addrs);
```

Comment:
you can write this more readable like this

```rust
println!(r#"
peer id: {}
listening addresses: {:?}
protocols: {:?}
protocol version: {}
observed addresses: {:?}
"#, l.peer_id, l.listen_addrs, l.protocols, l.protocol_version, l.observed_addrs);
```
Also I would recommend implementing `Display` for `Lookup` for this code

Reply:
I was told to remove my `Display` impl for Lookup.

Regardless, this needs to get cleaned, up. Going to deal with nice output in a follow up.